src/client/app/admin/user-management/admin-list-users.component.ts

```diff
@@ -115,24 +140,70 @@ export class AdminListUsersComponent {
 			this.pagingOpts.sortField = this.sortOpts.name.sortField;
 			this.pagingOpts.sortDir = this.sortOpts.name.sortDir;
 		}
+
+		if (cachedFilter.team) {
+			this.selectedTeam = new Team(cachedFilter.team._id, cachedFilter.team.name);
+		}
+
+		this.selectTeamsComponent.setSelectionInput(null, this.selectedTeam);
 	}
-	private loadUsers() {
+
+	resolveTeamNames() {
+		// Get unique list of team ids to query
+		let teamIds: string[] = _.uniq(_.flatMap(this.users, (user: TeamMember) => user.userModel.teams.map((t: any) => t._id)));
+
+		// Check to see if these ids have been cached
+		let idsToQuery = teamIds.filter((id: string) => !this.teamMap.hasOwnProperty(id));
+
+		// Retrieve data about unknown team ids
+		Observable.forkJoin(idsToQuery.map((id: string) => this.teamsService.get(id)))
+			.subscribe(
+				(results: any[]) => {
+					if (_.isArray(results)) {
+						// Cache the id to name mapping
+						results.forEach((r: any) => this.teamMap[r._id] = r.name);
+					}
+				},
+				(err: any) => {
+					// If unable to resolve teams, hide the team column
+					this.columns.teams.show = false;
+					this.alertService.addAlert(err.message);
+				});
+	}
+
+	loadUsers() {
 		let options: any = {};
-		this.adminService.cache.listUsers = {filters: this.filters, search: this.search, paging: this.pagingOpts};
-		this.adminService.search(this.getQuery(), this.search, this.pagingOpts, options)
-			.subscribe((result: any) => {
+
+		this.adminService.cache.listUsers = {
+			filters: this.filters,
+			search: this.search,
+			paging: this.pagingOpts,
+			team: this.selectedTeam,
+		};
+
+		let obs: Observable<Response> = (null != this.selectedTeam) ?
+			this.teamsService.searchMembers(this.selectedTeam._id, this.getQuery(), this.search, this.pagingOpts) :
+			this.adminService.search(this.getQuery(), this.search, this.pagingOpts, options);
+
+		obs.subscribe(
+			(result: any) => {
 				if (result && Array.isArray(result.elements)) {
-					this.users = result.elements.map((element: any) => {
-						return new User().setFromUserModel(element);
-					});
 					this.pagingOpts.set(result.pageNumber, result.pageSize, result.totalPages, result.totalSize);
+
+					// Set the user list
+					this.users = result.elements.map((element: any) => new TeamMember().setFromTeamMemberModel(null, element));
```

Comment:
would it make sense to update the teamsService and adminService functions called above to do some of the initial model construction like we've started doing in other services? then they could just return an array of TeamMembers. maybe the services could even populate team metadata so this component doesn't have to do all the work in resolveTeamNames